server/interceptor/interceptor.go

```diff
@@ -107,24 +109,24 @@ func getRPCInfo(req interface{}, fullMethod, quotaUser string) (*rpcInfo, error)
 		return nil, status.Errorf(codes.Internal, "cannot retrieve treeID from request: %T", req)
 	}
 
-	serviceName, methodName, err := parseFullMethod(fullMethod)
-	if err != nil {
-		return nil, err
-	}
-
 	var treeType trillian.TreeType
-	switch serviceName {
-	case "trillian.TrillianAdmin":
+	switch {
+	case isAdminRPC(req):
 		treeType = trillian.TreeType_UNKNOWN_TREE_TYPE // unrestricted
-	case "trillian.TrillianLog":
+	case isLogRPC(req):
 		treeType = trillian.TreeType_LOG
-	case "trillian.TrillianMap":
+	case isMapRPC(req):
 		treeType = trillian.TreeType_MAP
 	default:
-		return nil, status.Errorf(codes.Internal, "cannot determine treeType for request: %T", req)
+		return nil, status.Errorf(codes.Internal, "unmapped request type: %T", req)
 	}
 
-	readonly := strings.HasPrefix(methodName, "Get") || strings.HasPrefix(methodName, "List")
+	reqType := fmt.Sprintf("%T", req)
```

Comment:
I tried a bunch of variations of "isAdmin" and others before deciding on this one. The advantage here is that it doesn't need a lot of human intervention: if you add a new RPC, you add it to the appropriate list and that's it. There isn't a huge mapping of request to {treeID, treeType, readonly} that has to be written (and reviewed) by humans. We have a simple list of requests and a consistent set of rules, applied across all requests.